subcommand/common/common.go

```diff
@@ -75,10 +75,25 @@ func ConsulLogin(client *api.Client, bearerTokenFile, authMethodName, tokenSinkF
 		return fmt.Errorf("error logging in: %s", err)
 	}
 
-	// Write the token out to file with permissions so consul-k8s user can read.
-	payload := []byte(tok.SecretID)
-	if err := ioutil.WriteFile(tokenSinkFile, payload, 0444); err != nil {
+	if err := WriteFileWithPerms(tokenSinkFile, tok.SecretID, 0444); err != nil {
 		return fmt.Errorf("error writing token to file sink: %v", err)
 	}
 	return nil
 }
+
+// WriteFileWithPerms will write payload as the contents of the outputFile and set permissions.
+func WriteFileWithPerms(outputFile, payload string, mode os.FileMode) error {
+	// ioutil.WriteFile truncates existing files or if they do not exist writes them as FileMode(0666), but only if
+	// the file is currently writable. If the file exists it will already likely be read-only. Remove it.
+	if _, err := os.Stat(outputFile); err == nil {
+		err = os.Remove(outputFile)
+		if err != nil {
+			return fmt.Errorf("unable to delete existing token file: %s", err)
+		}
+	}
+	err := ioutil.WriteFile(outputFile, []byte(payload), os.FileMode(0666))
+	if err != nil {
+		return fmt.Errorf("unable to write file: %s", err)
+	}
```

Comment:
Can we use `0777` instead of `0666`? Shouldn't matter given we `os.Chmod()` it anyway right?
```suggestion
	if err := ioutil.WriteFile(outputFile, []byte(payload), os.ModePerm); err != nil {
		return fmt.Errorf("unable to write file: %s", err)
	}
```

Reply:
Great suggestions here and good catch on the os/ioutil changes.
I think going 0777 is fine as well.
Funny story: You are unable to apply suggested changes to the same line and all three of these touch (but dont change) the same lines so I have to do this all manually :D